man/org.freedesktop.login1.xml

```diff
@@ -549,8 +552,17 @@ node /org/freedesktop/login1 {
       is any.</para>
 
       <para><function>ListSessions()</function> returns an array of all current sessions. The structures in
-      the array consist of the following fields: session id, user id, user name, seat id, session object
-      path. If a session does not have a seat attached, the seat id field will be an empty string.</para>
+      the array consist of the following fields: <varname>session id</varname>, <varname>user id</varname>,
+      <varname>user name</varname>, <varname>seat id</varname>, and <varname>session object path</varname>.
+      If a session does not have a seat attached, the seat id field will be an empty string.</para>
+
+      <para><function>ListSessionsEx()</function> returns an array of all current sessions with more metadata
```

Comment:
Given it's new, should this also return the pid fd?

Reply:
I mean for the leader, other than the pid

Reply:
Hmm, I doubt if pidfd would show much value here. When we want to submit/query some processes, i.e. as input, pidfd is always the way to go, as the message might get dispatched much later, and the receiver needs to operate on the processes race-freely.

However, for results like this, which in most cases will be directly shown to the user, I don't think race is avoidable nor does it matter a lot. When building the format table, some sessions in the beginning might have even been stopped when we actually get to print it, or some new sessions have been started.

Reply:
As long as you are sure it will never be needed - as always, it's D-Bus so it cannot be changed later

Reply:
I am not sure it makes sense to return a myriad of fds in a list command. I think returning an fd on request is fine, but not just in a potentially huge dump of things. fds are still expensive in some contexts.